server/src/controllers/notification.controller.ts

```diff
@@ -0,0 +1,18 @@
+import { Controller, HttpCode, HttpStatus, Post } from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+import { AuthDto } from 'src/dtos/auth.dto';
+import { Auth, Authenticated } from 'src/middleware/auth.guard';
+import { NotificationService } from 'src/services/notification.service';
+
+@ApiTags('Notification')
+@Controller('notification')
+export class NotificationController {
+  constructor(private service: NotificationService) {}
+
+  @Post('test-email')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @Authenticated()
+  testEmailNotification(@Auth() auth: AuthDto) {
+    return this.service.handleTestEmailSetup({ id: auth.user.id });
```

Comment:
Feedback from the web notification is pretty bad. It tells you a test email has been sent, even though the smtp config is empty.

Why isn't `NotificationRepository.sendEmail` called directly? Then any errors can be relayed to the user.

Reply:
I am reworking to invoke those repository method directly

Reply:
That's much better already. However, I still think there's room for improvement. When an smtp error like `530 5.7.0 User not authenticated` occurs, it would be beneficial to show this error to troubleshoot the settings.